### ISSUE TITLE
PlusEq and other operations between QDenseReal and QDiagReal

### ISSUE THREAD

**Issue by mingruyang** [ITensor/ITensor#297]
Now if I use "delta(i, prime(i)) - T" where T is a dense ITensor(i, prime(i)), I will get "terminate called after throwing an instance of 'itensor::ITError'
  what():  doTask not defined for task PlusEQ and storage types QDenseReal QDiagReal". I think it would be useful to add this function. Or we can add a function "eye" to construct a dense identity ITensor similar to that for the Matrix.

**Comment by mtfishman** [comment]
Yes we should definitely have that operation, it just has not been very high up on the priority list.

One thing you could do right now is `toDense(delta(i,prime(i))) - T`.
